File: token-client-spring-demo/src/main/java/no/nav/security/token/support/demo/spring/client/DemoClient3.java
package no.nav.security.token.support.demo.spring.client;

import no.nav.security.token.support.demo.spring.config.DemoConfiguration;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

@Service
public class DemoClient3 {

    private final String url;
    private final RestTemplate restTemplate;

    public DemoClient3(@Value("${democlient3.url}") String url,
                       @DemoConfiguration.DemoClient3 RestTemplate restTemplate) {
        this.url = url;
        this.restTemplate = restTemplate;
    }

    public String ping() {
        return restTemplate.getForObject(url + "/ping", String.class);
    }
}

File: token-client-spring-demo/src/main/kotlin/no/nav/security/token/support/demo/spring/client/DemoClient3.kt
package no.nav.security.token.support.demo.spring.client

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Service
import org.springframework.web.client.RestClient.Builder
import org.springframework.web.client.body

@Service
class DemoClient3(@Value("\${democlient3.url}") url : String, builder : Builder) {

    private val client = builder.baseUrl(url).build()
    fun ping() = client.get()
        .uri { b -> b.path("/ping").build() }
        .retrieve()
        .body<String>()
}